src/main/java/com/microsoft/aad/msal4j/TokenCache.java

```diff
@@ -80,23 +79,40 @@ public void deserialize(String data) {
         this.appMetadata = deserializedCache.appMetadata;
     }
 
+    private static void mergeJsonObjects(JsonObject old, JsonObject update) {
+
+        for (Map.Entry<String, JsonElement> uEntry : update.entrySet())
+        {
+            String key = uEntry.getKey();
+            JsonElement uValue = uEntry.getValue();
+            if (!old.has(key)) {
+                if(!uValue.isJsonNull() &&
+                        !(uValue.isJsonObject() && uValue.getAsJsonObject().size() == 0)){
```

Comment:
Should this uValue.getAsJsonObject().size() != 0

Reply:
this expression means if 
value is not null and not (empty object)
add it save it